docs/rule/no-builtin-form-components.md

```diff
@@ -0,0 +1,65 @@
+# no-builtin-form-components
+
+Ember's built-in form components use two-way data binding, where the property as `@value` or `@checked` is mutated by user interaction. This goes against the Data Down Actions Up principle, goes against Glimmer Components’ intention to have immutable arguments, and is [discouraged by the Ember Core team](https://www.pzuraq.com/on-mut-and-2-way-binding/).
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+<Input />
+```
+
+```hbs
+<Textarea></Textarea>
+```
+
+## Migration
```

Comment:
There is another migration path, too.

Which is faster, and involves less binding.

See:
- https://tutorial.glimdown.com/7-form-data/1-automatic-binding
- and https://ember-primitives.pages.dev/7-forms/1-intro

Reply:
True. I've seen some of this before but never used these patterns exactly. Not sure I understand them well enough to write concisely about it.

How would you suggest the rule docs are structured?

Should we branch the suggestions based on their existing form type somehow or just present both (light/controlled) as options?

I guess ideally the cognitive load is fairly simple for users with simple forms but  🤷🏻

Reply:
Option 2 would be my preference.
You can copy from the above links 🎉

Reply:
I've made minor changes to incorporate your recommendations.

To be frank the docs on the ember primitives page are not easy to follow and don't clearly explain the light/controlled concepts so I don't really want to copy that over to this rule doc 😅 

For many cases they will be the ideal upgrade path but I think keeping an example of a low-level migration without addons is valuable for users to understand.

Reply:
can you open an issue on ember-primitives? docs should be easy to understand, and it's a bug that they're not https://github.com/universal-ember/ember-primitives/ thanks!!

Reply:
You got it – https://github.com/universal-ember/ember-primitives/issues/165